keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,541 @@
+
+## Design Details
+
+Use the same Kubelet Config as in linux.
+```
+type KubeletConfiguration struct {
+    ...
+    ShutdownGracePeriod metav1.Duration
+    ShutdownGracePeriodCriticalPods metav1.Duration
+}
+```
+
+Communication with service control manager will make use of golang.org/x/sys/windows/svc package,
+which is already included in [vendor](https://github.com/kubernetes/kubernetes/tree/release-1.19/vendor/golang.org/x/sys/windows)
+
+
+Termination of pods will make use of the existing
+[killPod](https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/kubelet/pod_workers.go#L292) function
+from the `kubelet` package and specify the appropriate `gracePeriodOverride` as
+necessary.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+The main logic of kill pod is the same as what linux did and there have been unit tests covered there.
+
+For Windows specific logic such as update registry keys, setting Windows service control handler, 
+It does not seem possible to cover with unit tests, so will cover with e2e tests.
+
+
+##### Integration tests
+
+There are efforts being worked on, but automated testing framework for Windows node level integration is not ready yet.
+Until then, we will cover all the scenerios with e2e tests
+
+##### e2e tests
+
+*   New E2E tests to validate Windows node graceful shutdown.
+    *   Shutdown grace period unspecified, feature is not active
+    *   Pod’s ExecStop and SIGTERM handlers are given gracePeriodSeconds for
+        case when gracePeriodSeconds <= kubeletConfig.ShutdownGracePeriod
+    *   Pod’s ExecStop and SIGTERM handlers are given
+        kubeletConfig.ShutdownGracePeriod for case when gracePeriodSeconds >
+        kubeletConfig.ShutdownGracePeriod
+
+### Graduation Criteria
+
+
+#### Alpha Graduation
+
+* Implemented the feature for Windows service only
+
+* Investigate how e2e tests can be implemented (e.g. may need to create fake
+  shutdown event)
+
+#### Alpha -> Beta Graduation
+
+* Addresses feedback from alpha testers
+* Sufficient E2E and unit testing
+
+#### Beta -> GA Graduation
+
+* Addresses feedback from beta
+* Sufficient number of users using the feature
+* Confident that no further API / kubelet config configuration options changes are needed
+* Close on any remaining open issues & bugs
+
+### Upgrade / Downgrade Strategy
+
+Before GA graduation, feature gates will be used to control the enable/disable of this enhacement. During all phase, 
+in order to enable this feature, the kubelet itself will need to run as a windows service.
+
+### Version Skew Strategy
+
+n/a
+
+## Production Readiness Review Questionnaire
```

Comment:
This KEP's markdown does not seem to be following the standard template.  Please update it to the regular template.

Reply:
updated with new template format